### DISCUSSION TITLE
Fork for AmneziaWG

### DISCUSSION THREAD

**Discussion by Angry-Engineer** [WGDashboard/WGDashboard#553] [Ideas]
Will be there any fork for amnezia wireguard?

**Comment by donaldzou** [comment]
Hi! Well there's no need for a fork. In our current `v4.2-dev` branch AmneziaWG is added to support recently and is stable. We're still figuring out to have it work with our Docker image. If you would like to, you can try it out :) The installation process is the same: clone, `./wgd.sh install` and run

It will check if `awg` and `awg-quick` exist, if not simply install them and restart WGDashboard :)

**Reply by it800** [reply]
Hi
  I installed 4.2-dev and met the following issue while trying to add new client in wgdashboard:
`Command 'awg set awg0 peer ХХХ allowed-ips 10.88.0.3/32 advanced-security off' returned non-zero exit status 1.`
I tried to execute it manually and got this error:
```
# awg set awg0 peer XXX allowed-ips 10.88.0.3/32 advanced-security off
Invalid argument: advanced-security
```

I'm using AWG installed as a kernel module. It looks like advanced-security option is not supported by AWG. Is there any way to make WGDashboard not to use it?

**Reply by DaanSelen** [reply]
Interesting, if you have suggestions, they are welcome. The bare-metal installation can use the kernel module, the docker solution uses the user-mode AWG binaries.

**Reply by it800** [reply]
I edited /etc/amnezia/amneziawg/WGDashboard/src/dashboard.py and removed the "advanced-security" option from the two lines of code where it was specified. It works fine now. I do not understand the need of that parameter because the settings of the clients derive advanced security options of Amnezia WG protocol from the main configuration.

**Reply by DaanSelen** [reply]
Sounds good, if you are open to it you can submit a PR to v4.2-dev. Otherwise I can take over if you'd like.

**Reply by it800** [reply]
Could you please take over this task? I'm not a programmer and can't submit a PR properly. Kindly note that this option might require deeper investigation, as I find it hard to believe that its support was added by mistake. However, if this option has been deprecated, then there's definitely no point in supporting it anymore.

**Reply by DaanSelen** [reply]
I'm picking this up With Donald.

**Reply by donaldzou** [reply]
> Could you please take over this task? I'm not a programmer and can't submit a PR properly. Kindly note that this option might require deeper investigation, as I find it hard to believe that its support was added by mistake. However, if this option has been deprecated, then there's definitely no point in supporting it anymore.

Hi! Just checked the source code of the kernel version, somehow `advance-security` is still in there.. are u using the GO version?

**Comment by dlangham547** [comment]
I have just downloaded the v4.2-Dev branch, installed it and also installed AmneziaWG. Is there anything else that I need to configure for WGDashboard to work with AmneziaWG as I don't see any settings available in the Dashboard itself?

**Comment by dlangham547** [comment]
Ok! Bit of an update... I'm running Ubuntu Server 24.04 and eventually found the config lol. I was then having errors around the amneziawg install which I hadn't spotted!!!
With Ubuntu Server 24.04 you need to edit the /etc/apt/sources.list.d/ubuntu.sources file adding in **deb-src** to the original as below, update the repositories and run a reinstall of amneziawg:

```
Types: deb deb-src
URIs: http://archive.ubuntu.com/ubuntu
Suites: noble noble-updates noble-backports
Components: main restricted universe multiverse
Signed-By: /usr/share/keyrings/ubuntu-archive-keyring.gpg

Types: deb deb-src
URIs: http://security.ubuntu.com/ubuntu
Suites: noble-security
Components: main restricted universe multiverse
Signed-By: /usr/share/keyrings/ubuntu-archive-keyring.gpg
```
Hope this helps if anyone else has an issue with 24.04

**Reply by DaanSelen** [reply]
> Ok! Bit of an update... I'm running Ubuntu Server 24.04 and eventually found the config lol. I was then having errors around the amneziawg install which I hadn't spotted!!! With Ubuntu Server 24.04 you need to edit the /etc/apt/sources.list.d/ubuntu.sources file adding in **deb-src** to the original as below, update the repositories and run a reinstall of amneziawg:
> 
> ```
> Types: deb deb-src
> URIs: http://archive.ubuntu.com/ubuntu
> Suites: noble noble-updates noble-backports
> Components: main restricted universe multiverse
> Signed-By: /usr/share/keyrings/ubuntu-archive-keyring.gpg
> 
> Types: deb deb-src
> URIs: http://security.ubuntu.com/ubuntu
> Suites: noble-security
> Components: main restricted universe multiverse
> Signed-By: /usr/share/keyrings/ubuntu-archive-keyring.gpg
> ```
> 
> Hope this helps if anyone else has an issue with 24.04

Good you found it! All credit to you, I am used to Debian so sorry for my lack of Ubuntu knowledge.

**Reply by dlangham547** [reply]
> > Ok! Bit of an update... I'm running Ubuntu Server 24.04 and eventually found the config lol. I was then having errors around the amneziawg install which I hadn't spotted!!! With Ubuntu Server 24.04 you need to edit the /etc/apt/sources.list.d/ubuntu.sources file adding in **deb-src** to the original as below, update the repositories and run a reinstall of amneziawg:
> > ```
> > Types: deb deb-src
> > URIs: http://archive.ubuntu.com/ubuntu
> > Suites: noble noble-updates noble-backports
> > Components: main restricted universe multiverse
> > Signed-By: /usr/share/keyrings/ubuntu-archive-keyring.gpg
> > 
> > Types: deb deb-src
> > URIs: http://security.ubuntu.com/ubuntu
> > Suites: noble-security
> > Components: main restricted universe multiverse
> > Signed-By: /usr/share/keyrings/ubuntu-archive-keyring.gpg
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Hope this helps if anyone else has an issue with 24.04
> 
> Good you found it! All credit to you, I am used to Debian so sorry for my lack of Ubuntu knowledge.

You can just run the below and it will add it in for you.
`sudo sed -i 's/^Types: deb$/Types: deb deb-src/' /etc/apt/sources.list.d/ubuntu.sources`

**Reply by hadi-dindar** [reply]
> Ok! Bit of an update... I'm running Ubuntu Server 24.04 and eventually found the config lol. I was then having errors around the amneziawg install which I hadn't spotted!!! With Ubuntu Server 24.04 you need to edit the /etc/apt/sources.list.d/ubuntu.sources file adding in **deb-src** to the original as below, update the repositories and run a reinstall of amneziawg:
> 
> ```
> Types: deb deb-src
> URIs: http://archive.ubuntu.com/ubuntu
> Suites: noble noble-updates noble-backports
> Components: main restricted universe multiverse
> Signed-By: /usr/share/keyrings/ubuntu-archive-keyring.gpg
> 
> Types: deb deb-src
> URIs: http://security.ubuntu.com/ubuntu
> Suites: noble-security
> Components: main restricted universe multiverse
> Signed-By: /usr/share/keyrings/ubuntu-archive-keyring.gpg
> ```
> 
> Hope this helps if anyone else has an issue with 24.04

did you install v.4.2  and amnziawg support inside this version ? work awg ?

**Reply by donaldzou** [reply]
For myself I did have AWG successfully installed on 22.04 with the official guide

**Comment by hadi-dindar** [comment]
@donaldzou  i install wgdashboard v4.2 on my vps 
also install amnziawg accordibg [this](https://docs.edisglobal.com/advanced-setup-guides/install-amneziawg-on-ubuntu-22_04/install-amneziawg-on-ubuntu-2204) 
now inside WGDasboard i can see amnziawg and i added a wg1 interface as amnziawg 
but i can not run it for this erorr

>ip link add wg1 type amneziawg Error: Unknown device type. Unable to access interface: Protocol not supported [#] ip link delete dev wg1 Cannot find device "wg1"
![Screenshot 2025-04-22 094250](https://github.com/user-attachments/assets/1baa6afb-929b-4d3b-964b-fe65303d49ce)

**Reply by DaanSelen** [reply]
If you can make it reproducable, please do so and send me a doc. I can check it out. Do the command awg-quick work if you do it manually?

**Reply by hadi-dindar** [reply]
> If you can make it reproducable, please do so and send me a doc. I can check it out. Do the command awg-quick work if you do it manually?

![Screenshot 2025-04-22 131539](https://github.com/user-attachments/assets/655bcd57-d394-4203-acba-f9781605766b)
 solved .tnx

**Reply by DaanSelen** [reply]
What was the issue and what was the solution?

**Reply by hadi-dindar** [reply]
> What was the issue and what was the solution?

problem was amnziawg kernel installing in linux . 
i treid install with this link
https://docs.edisglobal.com/advanced-setup-guides/install-amneziawg-on-ubuntu-22_04/install-wireguard-client-on-ubuntu-2204-and-connecting-to-the-server-using-amneziawg

**Reply by DaanSelen** [reply]
> > What was the issue and what was the solution?
> 
> problem was amnziawg kernel installing in linux . i treid install with this link https://docs.edisglobal.com/advanced-setup-guides/install-amneziawg-on-ubuntu-22_04/install-wireguard-client-on-ubuntu-2204-and-connecting-to-the-server-using-amneziawg

Okay if you have done things differently, please let us know so others can follow in the future.
